### DISCUSSION TITLE
How to know that the lease key has been deleted

### DISCUSSION THREAD

**Discussion by xiaocai2333** [etcd-io/etcd#15077] [General]
If I create a lease and attach it with a key, but I delete the key manually. How to let `LeaseKeepAlive` channel get it and give up renew? Can I only use the `Watch`?

**Comment by SimFG** [comment]
You can use the `TimeToLive` to check whether the lease have been attached to other keys when deleting the key manually. If no key attached to this lease, you can give up renew.
